Week 6/MyBookCollection/app/src/main/java/com/carol/mybookcollection/viewmodel/UserViewModel.kt

```diff
@@ -0,0 +1,30 @@
+package com.carol.mybookcollection.viewmodel
+
+import android.app.Application
+import android.content.Context
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.carol.mybookcollection.db.UserAccountDatabase
+import com.carol.mybookcollection.model.UserAccount
+import com.carol.mybookcollection.repository.UserRepository
+import kotlinx.coroutines.coroutineScope
+
+class UserViewModel(application: Application) : AndroidViewModel(application) {
+
+    private var repository: UserRepository = UserRepository(application)
+    private var getAllDatas: LiveData<List<UserAccount>>
+
+    init {
+
+        getAllDatas = repository.getAllData()!!
+    }
+
+    suspend fun insert(data: UserAccount) = coroutineScope {
```

Comment:
This is a nice try but I don't think it's quite what the assignment is asking for. It is always good to try because that's how to improve at code! See the solution for an example repo.